test/framework/integration_framework/fake_peer/fake_peer.cpp

```diff
@@ -135,8 +135,8 @@ namespace integration_framework {
       return *this;
     }
 
-    boost::optional<const BlockStorage &> FakePeer::getBlockStorage() const {
-      return {block_storage_ != nullptr, *block_storage_};
+    std::shared_ptr<BlockStorage> FakePeer::getBlockStorage() const {
```

Comment:
Please rework with optional and const ref.
 
Discussed offline. (we use optional when an object may absent. we use shared_ptr only for access management and we rely on the fact that the pointer is always initialized)